### PR TITLE
Add count methods to ELC's ReactiveElasticsearchClient.

### LINKED ISSUE #2749: ReactiveElasticsearchClient missing simple count() method
Is there a way to send a CountRequest without having an entity and repository?
What if I want to count number of documents in some index when having only index name?

### PR DESCRIPTION
**Pull request by maryantocinn**
Please let me know if i've missed anything as this is my first time contributing to an open source project.

- Implemented count methods using CountRequest directly with ELC's ReactiveElasticsearchClient, eliminating the need to initialize any entity or repository. 
- Refactored the existing doCount method in ReactiveElasticsearchTemplate to utilize the newly added count method.

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] **There is a ticket in the bug tracker for the project in our [issue tracker](https://github.com/spring-projects/spring-data-elasticsearch/issues)**. Add the issue number to the _Closes #issue-number_ line below
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Closes #2749

### CODE REVIEW

**Review by sothawo — CHANGES_REQUESTED** [review]
Thanks for your PR to add the `count` method to the `ReactiveElasticsearchClient`, this is all fine for the `ReactiveElasticsearchClient`, but we cannot use this in `org.springframework.data.elasticsearch.client.elc.ReactiveElasticsearchTemplate#doCount` - at least not without further investigation.

Why? Originally the `doCount` was implemented with a `CountRequest`( #341), the old one from the previous Elasticsearch libraries. We had to change this to a search query with a size of 0; you find a hint in this old issue: #1290, the count request does not consider the filters set on a query. or the diff in this old commit: https://github.com/spring-projects/spring-data-elasticsearch/commit/2cd18178e3b15bb1c531ef4cd59379a23b735c99#diff-8edbaefd20710bf00d7a492f1d368f3c9efbce5b4628afd4b43a72a8b28af17fL518-R294

Before going to a count request again, it would be necessary to investigate in that and see if it is safe to do this replacement.

But that I'd see in a diferent issue, and have this one just for adding the methods to the `ReactiveElasticsearchClient`.

**Comment by maryantocinn** [comment]
I see, thank you for the context. I'll leave the investigation of reverting to a `CountRequest` for `ReactiveElasticsearchTemplate#doCount` for a separate issue as it will need a deeper analysis of how the filters work.

For now, I'll revert all the changes made to `ReactiveElasticsearchTemplate` and focus this PR on adding the count methods to the `ReactiveElasticsearchClient` as initially intended.

File: src/main/java/org/springframework/data/elasticsearch/client/elc/ReactiveElasticsearchTemplate.java [hunk 1]
```diff
@@ -483,11 +484,10 @@ protected Mono<Long> doCount(Query query, Class<?> entityType, IndexCoordinates
 		Assert.notNull(query, "query must not be null");
 		Assert.notNull(index, "index must not be null");
 
-		SearchRequest searchRequest = requestConverter.searchRequest(query, routingResolver.getRouting(), entityType, index,
-				true);
+		CountRequest request = requestConverter.countRequest(query, entityType, index);
```
[the remark this answers is not in the document]
Comment (maryantocinn) [reply src/main/java/org/springframework/data/elasticsearch/client/elc/ReactiveElasticsearchTemplate.java:487]: Noted, will remove it.

File: src/main/java/org/springframework/data/elasticsearch/client/elc/RequestConverter.java [hunk 2]
```diff
@@ -1188,6 +1189,18 @@ public UpdateByQueryRequest documentUpdateByQueryRequest(UpdateQuery updateQuery
...
+	public <T> CountRequest countRequest(Query query, @Nullable Class<T> clazz, IndexCoordinates indexCoordinates) {
+
+		Assert.notNull(query, "query must not be null");
+		Assert.notNull(indexCoordinates, "index must not be null");
+
+		elasticsearchConverter.updateQuery(query, clazz);
+
+		return CountRequest.of(b -> b //
+				.query(getQuery(query, clazz)) //
+				.index(Arrays.asList(indexCoordinates.getIndexNames())));
```
[the remark this answers is not in the document]
Comment (maryantocinn) [reply src/main/java/org/springframework/data/elasticsearch/client/elc/RequestConverter.java:1201]: Thank you for pointing that out. Will look into how the `prepareSearchRequest` works, but for now will remove it as we cannot refactor `doCount` right now.

**Review by sothawo — APPROVED** [review]

**Pull request merged**